frontend/src/dataset_explorer/TableChartDataTable.tsx

```diff
@@ -0,0 +1,150 @@
+import React from "react";
+import {useTable, usePagination, useSortBy, Column} from "react-table";
+import { Cell, HeaderGroup, Row } from "react-table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import TableFooter from "@material-ui/core/TableFooter";
+import TablePagination from "@material-ui/core/TablePagination";
+import TableSortLabel from "@material-ui/core/TableSortLabel";
+import Paper from "@material-ui/core/Paper";
+import {MetricConfig} from '../data/MetricConfig';
+import { BreakdownVar, BREAKDOWN_VAR_DISPLAY_NAMES } from "../data/Breakdowns";
+import {Tooltip} from '@material-ui/core';
+import WarningRoundedIcon from "@material-ui/icons/WarningRounded";
+import TableContainer from '@material-ui/core/TableContainer';
+import Table from '@material-ui/core/Table';
+import {MetricId} from '../data/variableProviders';
+
+/** Corollary for TableHeader **/
+function TableHeader({ column }: { column: HeaderGroup<any> }) {
+  return (
+      <TableCell {...column.getHeaderProps(column.getSortByToggleProps())}>
+        {column.render("Header")}
+        <TableSortLabel
+            active={column.isSorted}
+            direction={column.isSortedDesc ? "desc" : "asc"}
+        />
+      </TableCell>
+  );
+}
+
+/** Corollary for TableHeaderGroup **/
+function TableHeaderGroup({ group }: { group: HeaderGroup<any> }) {
+  return (
+      <TableRow {...group.getHeaderGroupProps()}>
+        {group.headers.map((col, index) => (
+            <TableHeader column={col} key={index} />
+        ))}
+      </TableRow>
+  );
+}
+
+export interface DataTableProps {
+  data: Readonly<Record<string, any>>[];
+  metrics: MetricConfig[];
+  breakdownVar: BreakdownVar;
+}
+
+function TableChartDataTable(props: DataTableProps) {
+  const { data, metrics, breakdownVar } = props;
+  let columns = metrics.map((metricConfig) => {
+    return {
+      Header: metricConfig.fullCardTitleName,
+      accessor: metricConfig.metricId,
+    };
+  });
+  columns = [{
+    Header: BREAKDOWN_VAR_DISPLAY_NAMES[breakdownVar],
+    accessor: breakdownVar as MetricId,
+  }].concat(columns);
+
+  const memoCols = React.useMemo<Column<any>[]>(() => columns, []);
+  const memoData = React.useMemo(() => data, [data]);
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    page,
+    prepareRow,
+    gotoPage,
+    setPageSize,
+    state: {pageIndex, pageSize},
+  } = useTable(
+      {
+        columns: memoCols,
+        data: memoData,
+        initialState: {pageSize: 5},
+      },
+      useSortBy,
+      usePagination
+  );
+
+  function CustomTableRow({ row, }: { row: Row<any> }) {
```

Comment:
It took me a little while to figure out why the data row components were inside the main component and the header components were not. I see now that it's because you're using prepareRow here.

You might consider moving the header row inside as well just for consistency?